Automattic/wp-calypso
# Jetpack Google Analytics: fix the first GA settings update

## Proposed Changes

Jetpack no longer adds Google Analytics settings to the API response unless GA is active on the site.

This means that even after the module is activated, but the page is not yet refreshed (no settings re-fetch), Calypso does not know GA settings keys, and will not send them out upon saving.
That means that first attempt to save GA settings silently fails, as Calypso sends out empty list of settings to save.

This PR makes Calypso refresh settings after activating Google Analytics module so it would know about GA settings keys and be able to save them properly.

## Why are these changes being made?
pfwV0U-4g-p2
p8oabR-1ya-p2#comment-8110

## Testing Instructions

1. Connect Jetpack. If reusing a site, delete the option: `jetpack docker wp option delete jetpack_wga`
2. Go to "Jetpack -> Settings -> Traffic", find the "Google Analytics" card and purchase a plan.
3. After checkout, go back there and click on "Configure your Google Analytics settings".
4. Replace the `https://wordpress.com` with the Calypso Live URL from this PR.
5. In Calypso, find the "Google" card. Activate it by clicking "Add Google".
6. When the form shows up, enter a measurement ID (e.g. `G-12345`).
7. Click "Save settings". Check the Jetpack option and confirm it got saved: `jetpack docker wp option get jetpack_wga`.
8. Reload the page, confirm the measurement ID persists.

## Pre-merge Checklist
